mask/eolearn/mask/snow_mask.py

```diff
@@ -255,9 +262,18 @@ def _apply_first_pass(self, bands, ndsi, clm, dem, clm_temp):
                     snow_frac[date, nbin] = (
                         np.sum(snow_mask_pass1[date][in_dem_range_clear]) / dem_hist_clear_pixels[date, nbin]
                     )
-        return snow_mask_pass1, snow_frac, dem_edges
+        return snow_mask_pass1, snow_frac, dem_edges  # type: ignore[return-value]
```

Comment:
why is the ignore used here?

Reply:
error: Incompatible return value type (got "Tuple[ndarray[Any, dtype[Any]], Optional[ndarray[Any, dtype[floating[_64Bit]]]], Any]", expected
"Tuple[ndarray[Any, Any], ndarray[Any, Any], ndarray[Any, Any]]")  [return-value]


|name|got|expected|IMO|
|----|----|----|----|
|snow_mask_pass1|ndarray[Any, dtype[Any]]|np.ndarray|OK|
|snow_frac|Optional[ndarray[Any, dtype[floating[_64Bit]]]]|np.ndarray| idk |
|dem_edges|Any|np.ndarray|np.linspace|

Reply:
the types you assigned are clearly wrong since `snow_frac` is possibly `None`, you cannot ignore an error like this, it's the whole point of type checking

Reply:
done